.github/workflows/build-and-upload-release.yml

```
@@ -53,158 +53,11 @@ on:
-        uses: docker/setup-qemu-action@v1
-
-      - name: Build Binaries
-        run: |
-          go mod download
-          export GOARCH=arm64 && make targetallocator 
-          export GOARCH=amd64 && make targetallocator 
-
-      - name: Build Cloudwatch Agent Target Allocator Image and push to ECR
-        uses: docker/build-push-action@v4
-        if: steps.cached_binaries.outputs.cache-hit == false
-        with:
-          file: ./cmd/amazon-cloudwatch-agent-target-allocator/Dockerfile
-          context: ./cmd/amazon-cloudwatch-agent-target-allocator
-          push: true
-          tags: ${{ env.ECR_TARGET_ALLOCATOR_STAGING_REPO }}:${{ inputs.tag }}
-          platforms: linux/amd64, linux/arm64
-
-  bypass-info:
-    if: ${{ inputs.e2e-test-bypass-link != '' || inputs.e2e-test-bypass-approver != '' }}
-    runs-on: ubuntu-latest
-    steps:
-      - name: Echo bypass information
-        run: |
-          echo "**E2E test bypass requested**" >> $GITHUB_STEP_SUMMARY
-          echo "**Failing Test**: ${{ inputs.e2e-test-bypass-link }}" >> $GITHUB_STEP_SUMMARY
-          echo "**Approver**: ${{ inputs.e2e-test-bypass-approver }}" >> $GITHUB_STEP_SUMMARY
-
-  e2e-test:
-    if: ${{ inputs.e2e-test-bypass-link == '' && inputs.e2e-test-bypass-approver == '' }}
-    name: "Application Signals E2E Test"
-    needs: [MakeBinary]
-    uses: ./.github/workflows/application-signals-e2e-test.yml
+  BuildAndUpload:
+    uses: ./.github/workflows/build-and-upload.yml
     secrets: inherit
-    permissions:
-      id-token: write
-      contents: read
     with:
-      tag: ${{ inputs.tag }}
-
-  push-release-ecr:
-    if: ${{ inputs.release }}
-    needs: [MakeTABinary, e2e-test]
-    permissions:
-      id-token: write
-      contents: read
-    runs-on: ubuntu-latest
-    steps:
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v2
-        with:
-          role-to-assume: ${{ env.AWS_ASSUME_ROLE }}
-          aws-region: us-west-2 # Uploads only to us-west-2 for now
-
-      - name: Login ECR
-        id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
-
-      - name: Push image to release ECR
-        run: |
-          docker buildx imagetools create \
-          -t ${{ env.ECR_OPERATOR_RELEASE_IMAGE }} \
-          ${{ env.ECR_OPERATOR_STAGING_REPO }}:${{ inputs.tag }}
-
-      - name: Push image to TA release ECR
-        run: |
-          docker buildx imagetools create \
-          -t ${{ env.ECR_TARGET_ALLOCATOR_RELEASE_REPO}} \
-          ${{ env.ECR_TARGET_ALLOCATOR_STAGING_REPO }}:${{ inputs.tag }}
\ No newline at end of file
+      release: ${{inputs.release}}
+      tag: ${{inputs.tag}}
+      e2e-test-bypass-link: ${{inputs.e2e-test-bypass-link}}
+      e2e-test-bypass-approver: ${{inputs.e2e-test-bypass-approver}}
```

We need to implement the bypass logic when we call the next workflow.
```suggestion
      e2e-test-bypass-approver: ${{inputs.e2e-test-bypass-approver}}
      test-image-before-upload: ${{ inputs.e2e-test-bypass-link == '' && inputs.e2e-test-bypass-approver == '' }}
```